components/ShippingRates/index.vue

```diff
@@ -0,0 +1,159 @@
+          :icon="buttonConfig.icon"
+          :label="buttonConfig.text"
+          :loading="isLoading"
+          @click="buttonConfig.action"
+        />
+      </div>
+    </template>
+
+    <UTable
+      v-if="isEditMode"
+      :columns="[
+        { key: 'index' },
+        { key: 'name', label: 'Name' },
+        { key: 'price', label: 'Price (USD)' },
+      ]"
+      :rows="shippingRatesTableRows"
+    >
+      <template #name-data="{ row }">
+        <div class="flex flex-col gap-[8px] items-start">
+          <span class="text-center">en: {{ row.name.en }}</span>
+          <span class="text-center">zh: {{ row.name.zh }}</span>
+        </div>
+      </template>
+      <template #price-data="{ row }">
+        <span class="text-center">{{ row.price }}</span>
+      </template>
+    </UTable>
+
+    <div v-if="isViewMode" class="px-[24px] py-[12px]">
+      <UFormGroup
+        label="Physical Goods"
+        :ui="{ label: { base: 'font-mono font-bold' } }"
+      >
+        <div class="flex flex-col gap-[16px]">
+          <UCheckbox
+            v-model="hasShipping"
+            name="hasShipping"
+            label="Includes physical good that requires shipping"
+            :disabled="!shippingInfo.length"
+          />
+          <UAlert
+            v-if="!shippingInfo.length"
+            icon="i-heroicons-face-frown-solid"
+            color="yellow"
+            variant="solid"
+            title="Please set the shipping options first to enable this feature."
+          />
+        </div>
+      </UFormGroup>
+    </div>
+    <ShippingRatesInfoModal
+      v-if="isSippingModalOpened"
+      v-model="isSippingModalOpened"
+      :mode="mode"
+      :shipping-info="shippingInfo"
+      @on-update-shipping-rates="
+        (value) => emit('on-update-shipping-rates', value)"
+    />
+  </UCard>
+</template>
+
+<script setup lang="ts">
+type ModeType = 'edit' | 'view';
+const NEW_LISTING_PAGES = {
+  BOOK: 'nft-book-store-new',
+  COLLECTION: 'nft-book-store-collection-new'
+}
+const props = defineProps({
+  isLoading: {
+    type: Boolean,
+    default: false
+  },
+  mode: {
+    type: String as PropType<ModeType>,
+    default: 'edit'
+  },
+  shippingInfo: {
+    type: Array,
+    default: () => []
+  },
+  modelValue: {
+    type: Boolean
+  }
+})
+const route = useRoute()
+const router = useRouter()
+const routerName = route.name
+const emit = defineEmits(['update:modelValue', 'on-update-shipping-rates'])
+const isSippingModalOpened = ref<Boolean>(false)
+
+const hasShipping = ref(props.modelValue)
+watch(() => props.modelValue, (newValue) => {
+  hasShipping.value = newValue
+})
+watch(hasShipping, (hasShipping) => {
+  emit('update:modelValue', hasShipping)
+})
+const isEditMode = computed(() => !!(props.mode === 'edit'))
+const isViewMode = computed(() => !!(props.mode === 'view'))
+const isNewListingPage = computed(() => Object.values(NEW_LISTING_PAGES).includes(routerName as string))
```

Comment:
I think this is better set as a boolean prop since we might have new `new` pages in the future, setting them from the caller is simpler to organize

Reply:
[♻️ Use boolean prop to check if is new listing page](https://github.com/likecoin/nft-book-press/pull/88/commits/3659134cba4b56555075a96f710b4e15f2c921ce)